MaxText/layers.py

```diff
@@ -123,8 +127,22 @@ def dot_product_attention(query: Array,
   query = LayerNorm(dtype=dtype, name='query_layer_norm', kernel_axes = ('heads',))(query)
   key = LayerNorm(dtype=dtype, name='key_layer_norm', kernel_axes = ('heads',))(key)
 
-  # `attn_weights`: [batch, num_heads, q_length, kv_length]
-  attn_weights = jnp.einsum('bqhd,bkhd->bhqk', query, key)
+  # QK Product, a.k.a `attn_weights`: [batch, num_heads, q_length, kv_length]
+  if not cfg.int8_training:
+    attn_weights = jnp.einsum('bqhd,bkhd->bhqk', query, key)
+  else:
+    aqt_cfg = maxtext_sweeps.sweep1(
```

Comment:
I don't know what sweep1 is but I have a feeling it shouldn't be here?

Also this might conflict with FlashAttention in which you need to rebase on top of that.

Reply:
Finally can this be encapsulated in a function that computes attention? Because we're scattering the chaos when folks try to read the code.

Reply:
Renamed sweep1 to `quantization_config`. However I couldn't think of a quick way to better encapsulate the aqt computation -- possibly in the future we can have an aqt einsum wrapper that takes in
* The einsum equation (e.g. ab, bc -> ac)
* The inputs
* The aqt config
But that takes some refactoring and is unclear to me its any simpler than what exists

Reply:
After discussing with @rwitten offline we have moved each int8 einsum and dot general into their own methods so users may instead simply read only the method invocation instead of the method definition.

Future work may include create a parametrized structure of aqt_config options (a dict hierarchy in base.yml) so that the three aqt configs can be created from a single method